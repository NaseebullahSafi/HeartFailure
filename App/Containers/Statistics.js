import React, {Component} from 'react';
import {StyleSheet, Text, View} from 'react-native';
import ChartView from 'react-native-highcharts';
import Chart from "./Chart";

export default class Statistics extends Component {
  constructor(props) {
    super(props);

    this.state = {
      type: "Day"
    }
  }

  componentDidMount() {

  }

  render() {
    return (
      <View style={styles.container}>
        <View style={styles.calendarBtnContainer}>
          <Text style={styles.calendarBtn}>Month</Text>
          <Text style={styles.calendarBtn}>Day</Text>
          <Text style={styles.calendarBtn}>Year</Text>
        </View>
        <Chart style={styles.chartContainer} type={"day"}/>
        <View style={styles.calendarBtnContainer}>
          <Text style={styles.calculationContainer}>Min</Text>
          <Text style={styles.calculationContainer}>Max</Text>
          <Text style={styles.calculationContainer}>Avg</Text>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    flexDirection:'column',
    justifyContent: 'space-between'
  },
  calendarBtnContainer: {
    flexDirection: 'row',
    justifyContent: 'center'
  },
  calendarBtn: {
    flex: 1,
    marginTop: 30,
    marginBottom: 30,
    marginRight: 20,
    marginLeft: 20,
    textAlign: 'center',
    padding: 10,
    fontWeight: 'bold',
    color: 'white',
    borderRadius: 100 / 2,
    backgroundColor: '#c8d6dc',
    elevation: 5
  },
  chartContainer: {
    // flex: 20
  },
  calculationContainer: {
    flex: 1,
    marginBottom: 30,
    marginRight: 20,
    marginLeft: 20,
    textAlign: 'center',
    padding: 10,
    fontWeight: 'bold',
    color: '#909aae',
    fontSize: 19
  }
});
