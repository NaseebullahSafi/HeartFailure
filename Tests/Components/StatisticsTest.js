// need to import React itself
import React from 'react'

// everything fro react native
import 'react-native'

// required for mocking modules
import Enzyme,{ shallow } from 'enzyme'

// needed to pass components to the renderer
import renderer from 'react-test-renderer'

// adapter is needed for this react version
import Adapter from 'enzyme-adapter-react-16';

// the component that will be tested
import Statistics from "../../App/Containers/Statistics";

Enzyme.configure({ adapter: new Adapter() });

describe('Testing <Statistics /> component', () => {
  // create an image of the component, and match it with the snapshot
  it('renders as expected', () => {
    const tree = renderer.create(
      <Statistics />
    ).toJSON();
    expect(tree).toMatchSnapshot()
  });

  it('Should create a chart for the day', () => {
    const wrapper = shallow(<Statistics handleSubmit={jest.fn()} />);
    const inst = wrapper.instance().DayConfig()
    expect(inst)
  });

  it('Should create a chart for the month', () => {
    const wrapper = shallow(<Statistics handleSubmit={jest.fn()} />);
    const inst = wrapper.instance().MonthConfig()
    expect(inst)
  });

  it('Should create a chart for the year', () => {
    const wrapper = shallow(<Statistics handleSubmit={jest.fn()} />);
    const inst = wrapper.instance().YearConfig()
    expect(inst)
  });

  it('Should render the component from render() hook', () => {
    const wrapper = shallow(<Statistics handleSubmit={jest.fn()} />);
    const inst = wrapper.instance().render();
    expect(inst)
  });
});

