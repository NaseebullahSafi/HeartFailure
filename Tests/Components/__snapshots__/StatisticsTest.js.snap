// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Statistics /> component renders as expected 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "space-between",
    }
  }
>
  <RCTScrollView
    showsVerticalScrollIndicator={false}
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#E67D8F",
              "borderRadius": 50,
              "elevation": 5,
              "flex": 1,
              "marginBottom": 30,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 30,
              "maxWidth": 100,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontWeight": "bold",
                  "padding": 10,
                  "textAlign": "center",
                },
                Object {
                  "color": "white",
                },
              ]
            }
          >
            Day
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "marginBottom": 30,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 30,
              "maxWidth": 100,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontWeight": "bold",
                  "padding": 10,
                  "textAlign": "center",
                },
                Object {
                  "color": "#909aae",
                },
              ]
            }
          >
            Month
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flex": 1,
              "marginBottom": 30,
              "marginLeft": 20,
              "marginRight": 20,
              "marginTop": 30,
              "maxWidth": 100,
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontWeight": "bold",
                  "padding": 10,
                  "textAlign": "center",
                },
                Object {
                  "color": "#909aae",
                },
              ]
            }
          >
            Year
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "height": 300,
            "width": 750,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <RCTWebView
            allowsInlineMediaPlayback={undefined}
            automaticallyAdjustContentInsets={false}
            bounces={undefined}
            contentInset={undefined}
            dataDetectorTypes={undefined}
            decelerationRate={undefined}
            injectedJavaScript={undefined}
            mediaPlaybackRequiresUserAction={undefined}
            messagingEnabled={false}
            onLoadingError={[Function]}
            onLoadingFinish={[Function]}
            onLoadingStart={[Function]}
            onMessage={[Function]}
            onShouldStartLoadWithRequest={undefined}
            scalesPageToFit={false}
            scrollEnabled={false}
            source={
              Object {
                "baseUrl": "web/",
                "html": "<html>
                    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, user-scalable=0\\" />
                    <style media=\\"screen\\" type=\\"text/css\\">
                    #container {
                        width:100%;
                        height:100%;
                        top:0;
                        left:0;
                        right:0;
                        bottom:0;
                        position:absolute;
                        user-select: none;
                        -webkit-user-select: none;
                    }
                    </style>
                    <head>
                        <script src=\\"https://code.jquery.com/jquery-2.1.4.min.js\\"></script>
                        <script src=\\"https://code.highcharts.com/highcharts.js\\"></script>
                        
                        
                        <script src=\\"https://code.highcharts.com/modules/exporting.js\\"></script>
                        <script>
                        $(function () {
                            Highcharts.setOptions(undefined);
                            Highcharts.chart('container', {title: {text: \\"\\", style: {display: \\"none\\"}}, xAxis: {title: {text: \\"\\"}, type: \\"datetime\\", dateTimeLabelFormats: {day: \\"%e of %b\\"}, lineWidth: 0, minorGridLineWidth: 0, minorTickLength: 0, tickLength: 0, lineColor: \\"transparent\\", labels: {style: {color: \\"#959bad\\"}}}, yAxis: {title: {text: \\"\\"}, labels: {style: {color: \\"#959bad\\"}}, gridLineColor: \\"rgba(188, 202, 208, 0.5)\\"}, legend: {enabled: false}, exporting: {enabled: false}, credits: {enabled: false}, plotOptions: {series: {color: \\"#E67D8F\\", lineWidth: 2, shadow: true}, line: {marker: {enabled: true}}}, series: [{type: \\"spline\\", data: [29.9, 71.5, 106.4, 129.2, 144, 176, 135.6, 148.5, 216.4, 194.1, 95.6, 54.4], pointStart: 1262304000000, pointInterval: 86400000}]}           );
                        });
                        </script>
                    </head>
                    <body>
                        <div id=\\"container\\">
                        </div>
                    </body>
                </html>",
              }
            }
            style={
              Array [
                Object {
                  "flex": 1,
                },
                Object {
                  "backgroundColor": "#ffffff",
                },
                Object {
                  "height": 300,
                  "width": 750,
                },
              ]
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "column",
              "marginBottom": 30,
              "marginLeft": 20,
              "marginRight": 20,
              "padding": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#909aae",
                "fontSize": 19,
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Min
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#909aae",
                  "fontSize": 19,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                Object {
                  "color": "#d0d4db",
                  "fontSize": 29,
                  "marginTop": 10,
                },
              ]
            }
          >
            50
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
              "marginBottom": 30,
              "marginLeft": 20,
              "marginRight": 20,
              "padding": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#909aae",
                "fontSize": 19,
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Max
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#909aae",
                  "fontSize": 19,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                Object {
                  "color": "#d0d4db",
                  "fontSize": 29,
                  "marginTop": 10,
                },
              ]
            }
          >
            150
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
              "marginBottom": 30,
              "marginLeft": 20,
              "marginRight": 20,
              "padding": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#909aae",
                "fontSize": 19,
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Avg
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#909aae",
                  "fontSize": 19,
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
                Object {
                  "color": "#d0d4db",
                  "fontSize": 29,
                  "marginTop": 10,
                },
              ]
            }
          >
            112
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
